data/model-db/generate_historical_data.R

```diff
@@ -0,0 +1,778 @@
+try(if(any(is.na(annual_redd$year)))
+  stop("Missing Year! Please fix!"))
+
+gcs_upload(annual_redd,
+           object_function = f,
+           type = "csv",
+           name = "model-db/annual_redd.csv",
+           predefinedAcl = "bucketLevel")
+
+# passage_counts ----------------------------------------------------------
+gcs_get_object(object_name = "jpe-model-data/upstream_passage.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/model-data/upstream_passage.csv",
+               overwrite = TRUE)
+passage_raw <- read_csv("data/model-data/upstream_passage.csv")
+
+passage <- passage_raw |> 
+  # remove missing dates
+  filter(!is.na(date)) |> 
+  mutate(date = case_when(!is.na(time) ~ ymd_hms(paste0(date,time)),
+                           T ~ ymd_hms(paste0(date, " 00:00:00")))) |> 
+  rename(hours_sampled = hours) |> 
+  select(-c(time, viewing_condition, spawning_condition, jack_size, ladder, 
+            flow, temperature, comments)) |> 
+  # survey_location_id
+  mutate(stream = tolower(stream),
+         reach = NA,
+         sex = ifelse(is.na(sex), "not recorded", sex),
+         passage_direction = ifelse(is.na(passage_direction), "not recorded", passage_direction)) |> 
+  left_join(survey_location, by = c("stream", "reach")) |>
+  select(-c(stream, reach, description)) |>
+  rename(survey_location_id = id) |>
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description) |> 
+  # sex_id
+  left_join(sex, by = c("sex" = "definition")) |> 
+  rename(sex_id = id) |> 
+  select(-sex, -description) |> 
+  #direction_id
+  left_join(direction, by = c("passage_direction" = "definition")) |> 
+  rename(direction_id = id) |> 
+  select(-passage_direction, -description)
+
+try(if(any((unique(passage$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(passage$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any((unique(passage$sex_id) %in% sex$id) == F))
+  stop("Missing Sex ID! Please fix!"))
+try(if(any((unique(passage$direction_id) %in% direction$id) == F))
+  stop("Missing Direction ID! Please fix!"))
+try(if(any(is.na(passage$date)))
+  stop("Missing Date! Please fix!"))
+  
+gcs_upload(passage,
+           object_function = f,
+           type = "csv",
+           name = "model-db/passage.csv",
+           predefinedAcl = "bucketLevel")
+# passage_estimates -------------------------------------------------------
+gcs_get_object(object_name = "standard-format-data/standard_adult_passage_estimate.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_adult_passage_estimate.csv",
+               overwrite = TRUE)
+passage_estimates_raw <- read_csv("data/standard-format-data/standard_adult_passage_estimate.csv")
+
+passage_estimate <- passage_estimates_raw |> 
```

Comment:
Two NA `passage_estimate` values have info on `adipose_clipped` and `run_id`. We might just want to remove these for the time being. Or create an issue to look into them